exp/api/v1alpha3/awsfargateprofile_types.go

```diff
@@ -0,0 +1,162 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha3
+
+import (
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	infrav1 "sigs.k8s.io/cluster-api-provider-aws/api/v1alpha3"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1alpha3"
+	"sigs.k8s.io/cluster-api/errors"
+)
+
+const (
+	// FargateProfileFinalizer allows the controller to clean up resources on delete
+	FargateProfileFinalizer = "awsfargateprofile.infrastructure.cluster.x-k8s.io"
+)
+
+var (
+	// DefaultEKSFargateRole is the name of the default IAM role to use for fargate
+	// profiles if no other role is supplied in the spec and if iam role creation
+	// is not enabled. The default can be created using clusterawsadm or created manually
+	DefaultEKSFargateRole = fmt.Sprintf("eks-fargate%s", infrav1.DefaultNameSuffix)
+)
+
+// FargateProfileSpec defines the desired state of FargateProfile
+type FargateProfileSpec struct {
+	ClusterName string `json:"clusterName,omitempty"`
```

Comment:
Is this needed? Could the cluster name come from the `AWSManagedControlPlane` instead?

Reply:
At the moment `Spec.ClusterName` is how I get at the `AWSManagedControlPlane`:

https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/eceb96da3384769ac7c3aefa70c628c7efa51145/exp/controllers/awsfargatepool_controller.go#L89

Reply:
I've added the missing description to the field